### ISSUE TITLE
chromebook_snow: status

### ISSUE THREAD

**Issue by hexdump0815** [hexdump0815/imagebuilder#78]
## notes on snow running 5.18.1:

here is list of working and non working features on the snow chromebooks.

### working
- wifi
- bluetooth
- builtin display
- usb (the usb3 port is disabled)
- builtin speaker
- energy: suspend/resume

### untested
- hw video encode/decode (most probably not working)
- builtin mic (might work)
- external display (might work)

### broken
- gpu: only partially useable using the legacy mali driver and blob libraries in some special setups (it might be possible to get the open source panfrost driver somewhat working)
- energy: dpms
- usb3 port (disabled as it seems to break suspend/resume iirc)
- webcam (seems to be usb, but is not detected - maybe related to usb3 probems)

### installation
coming soon

### problems
- display power management not working and thus disabled by default
- booting via u-boot from usb does not work, via sd-card it should work

**Comment by jenneron** [comment]
> builtin mic (might work)

https://gitlab.com/postmarketOS/pmaports/-/blob/master/device/testing/linux-postmarketos-exynos5/0023-HACK-ARM-dts-exynos-snow-disable-HDMI-audio.patch

> webcam (seems to be usb, but is not detected - maybe related to usb3 probems)

https://dpaste.com/4X5Z7M7PL

**Comment by hexdump0815** [comment]
@jenneron - thanks a lot for those hints! - i really appreciate that you let me know of those changes as i was not aware of them yet ... looks like i'll have a closer look at all the things you already did for pmos on snow and peach and will most probably take over quite a few of those patches in my next snow/peach round which is currently planned for when v6.0 or maybe v6.1 is out (or if i find some time before) ... its definitely on my todo list for snow/peach now ... even if those changes get partially upstreamed (which is very nice as well) i'll apply them then as long as they are not in yet and drop them as soon as they are in the kernel sources

btw. can you confirm the problems a saw with suspend/resume breaking (i think it was randomly) on snow when the usb3 port gets enabled? i did not test it with usb3 enabled on the recent kernels, but the problem was there for quite a while for me ...

once more a lot of thanks and keep up all your good work - best wishes - hexdump

**Comment by jenneron** [comment]
> can you confirm the problems a saw with suspend/resume breaking (i think it was randomly) on snow when the usb3 port gets enabled?

@hexdump0815 usb 3.0 and deep suspend work in pmOS (5.16.1). Although this version (and many other actually, 5.19.0 is broken too, but 5.18.5 is not affected) has generic issues with camera on armv7

Speaking of upstreaming, I would like to get hp chromebook (spring) firstly and make corresponding patches for it. However it's quite possible I won't manage to get it for reasonable price and will just send patches without it

**Comment by hexdump0815** [comment]
@jenneron - ok, will give usb3 on newer kernels a try again then

i have a spring here and i think there is quite a bit more work required for it as it is not really working too well with mainline: for u-boot usb is not working (and spring does not have an sd card slot) and for the kernel the display is broken due to deferred devices and maybe more (i think there are quite a few things missing in the dtb)

**Comment by RaylynnKnight** [comment]
I have a Spring v5 (which I documented one [https://wikidevi.wi-cat.ru/Samsung_Chromebook_(XE303C12) WikiDevi].  Will be attempting to build and updated kernel with above patches soon and will post results here.

**Comment by hexdump0815** [comment]
@RaylynnKnight - spring or snow v5?

**Comment by RaylynnKnight** [comment]
@hexdump0815 - That should have been Snow v5 (i.e. Samsung Chromebook XE303C12). I've now edited comment.

**Comment by OscilloBecher** [comment]
Hello,
First, i am french .... i speak a special english ...
Thanks for all 
I have daisy snow model of chromebook et i try "230220-01 - debian bookworm image for the snow chromebook - samsung xe303c12", all seems fine, i see the login panel ! but could you say to me what i choose  for username and password (i tried "root" and "toor" ... dont match...)
Thanks

**Comment by hexdump0815** [comment]
@OscilloBecher - happy to hear that it seems to work well for you - the default username/password is linux/changeme - see https://github.com/hexdump0815/imagebuilder/blob/main/doc/first-steps.md (linked from the main readme)

good luck and best wishes - hexdump

**Comment by hexdump0815** [comment]
to have this information here as well: @jenneron from postmarketos bisected the broken mesa support and added an issue on mesa and maybe the gpu will work again soon with mainline kernel and mesa with a bit of luck - see: https://gitlab.freedesktop.org/mesa/mesa/-/issues/10558

**Comment by rtd1250** [comment]
I have tried it out, and yes it looks like gpu acceleration is working once again, with reasonable performance. I couldn't get it to work under the debian bookworm image (likely something kernel related, I made sure to remove the blacklisted modules from modprobe.d but didn't troubleshoot past that), however it works on arch linux arm. There seem to be some regressions compared to old mesa, though - desktop acceleration works fine apart from flickering cursor, but glxgears isn't displaying correctly while in old mesa it was fine. Still glad to see that work is still being done for this laptop, now the corruptions are gone

**Comment by jenneron** [comment]
>  There seem to be some regressions compared to old mesa

which exactly?

> though - desktop acceleration works fine apart from flickering cursor

wayland works fine. flickering cursor is not a big problem in terms of x11

> but glxgears isn't displaying correctly while in old mesa it was fine

i think it is being bisected (not by me, this time by another person)

**Comment by jenneron** [comment]
@rtd1250 can you run `glmark2` in x11 with default settings and tell us the score? I'm wondering if performance is different on upstream kernel. We need kernel hacks for T620 to make it performant enough, and I've been testing T600 with the same kernel

**Comment by rtd1250** [comment]
Arch ARM's kernel is sadly heavily outdated nowadays due to lack of maintenance so the results won't be very meaningful. I'm starting to look for ways to run a more modern kernel, if you've got a preferred kernel or other setup for this laptop I'd much appreciate if you shared it here since I'm not that much knowledgeable with this hardware.

I've noticed the exynos5-mainline project so I could try using that if it's suitable for use

Still, I'll try out running benchmarks on 5.5 as that's what is currently installed but I'm pretty sure that some of the problems I'm facing are definitely caused by the kernel itself.

**Comment by jenneron** [comment]
> if you've got a preferred kernel or other setup

Use kernel from postmarketOS
Right now it is 6.1.56 LTS
kernel: https://gitlab.com/exynos5-mainline/linux/-/archive/v6.1.56-exynos5-lts/linux-v6.1.56-exynos5-lts.tar.gz
config: https://gitlab.com/postmarketOS/pmaports/-/blob/b7427191bf6b5315c4eeb3815f905e8a0d8c25c3/device/community/linux-postmarketos-exynos5/config-postmarketos-exynos5.armv7

Unfortunately, one of USB ports doesn't work with this kernel right now

**Comment by jenneron** [comment]
The feature status with it is at https://wiki.postmarketos.org/wiki/Samsung_Chromebook_(google-snow)

**Comment by rtd1250** [comment]
I've managed to build the pmos kernel and get it running, it is a huge improvement compared to the old 5.5.7 arch kernel - KDE Wayland now works fine and X11 has also seen a performance increase - this combined with recent mesa changes results in the best desktop experience I've had on snow, KDE's webkit browser works smoothly too
![Screenshot_20240213_192906](https://github.com/hexdump0815/imagebuilder/assets/47224354/5f9b14ce-b5fd-4c9c-b43b-98a7d384a759)


In attachment you'll find the results of glmark2, only difference between them is the kernel, desktop environment was twm since performance is ~20 points worse under kde
[glmark2-results-snow.tar.gz](https://github.com/hexdump0815/imagebuilder/files/14271260/glmark2-results-snow.tar.gz)

**Comment by Salahd94** [comment]
Hi @rtd1250 

I'm encountering an issue while trying to burn a copy of Snow to a flash drive using Etcher. It gave me an error saying the boot partition was not found. Can you provide guidance on how to solve this problem?

Also, I'm interested in the improvements you've made compared to the old 5.5.7 version. Could you share more details on these enhancements and the process behind them?

Thanks!

**Comment by jenneron** [comment]
>  It gave me an error saying the boot partition was not found. 

do you mean on postmarketOS?

> Can you provide guidance on how to solve this problem?

use sd card instead of usb

**Comment by Salahd94** [comment]
Hi @jenneron, @rtd1250

Thank you for your suggestion to use an SD card instead of a USB drive. It worked perfectly, and I was able to resolve the boot partition error. I appreciate your help!

I have a couple more questions if you don't mind:
1. Could you please guide how to add the library of GPG on postmarketOS?
2. Additionally, I'm interested in installing KDE Plasma on postmarketOS or changing X11 to KDE Plasma. Could you assist me with that?

Thank you again for your assistance!

**Comment by rtd1250** [comment]
> 1. Could you please guide how to add the library of GPG on postmarketOS?
You may install packages with apk, links to Alpine/postmarketos package lists are also included here:
https://wiki.postmarketos.org/wiki/Terminal_cheat_sheet

> 2. Additionally, I'm interested in installing KDE Plasma on postmarketOS or changing X11 to KDE Plasma. Could you assist me with that?
Either install postmarketos-ui-plasma-desktop or flash an official image:
https://images.postmarketos.org/bpo/edge/google-snow/

To be fair I'm yet to try out pmOS with panfrost-accelerated desktop myself, I'll happily test it now that T600 support has landed in upstream mesa

**Comment by jenneron** [comment]
> It worked perfectly, and I was able to resolve the boot partition error. I appreciate your help!

usb booting should work with recent pmOS images, it has been fixed

**Comment by TheNathan0** [comment]
@rtd1250, not sure if this affects all Midgard v4 GPUs or just the T620 series of Mali GPUs, but do you get occasional crashes/freezes in KDE when you try to switch tabs or place your cursor on the top left corner of the screen?  I have a device with a T628 Mali GPU and for some reason on almost every distro (including postmarketOS) does placing the cursor on the top left corner of the screen or just trying to switch windows makes the desktop environment either crash or freeze; on X11, I believe it usually automatically recovers fairly quickly while on Wayland does it require some more manual work (otherwise usually you'll be stuck on a black screen with a cursor); other than that, KDE is stable on both Wayland and X11.  On Arch Linux ARM, this problem seems to be either nonexistent or just rarely(?) occurs but some strange lag spike occurs instead.

**Comment by rtd1250** [comment]
> @rtd1250, not sure if this affects all Midgard v4 GPUs or just the T620 series of Mali GPUs, but do you get occasional crashes/freezes in KDE when you try to switch tabs or place your cursor on the top left corner of the screen?

I didn't encounter these freezes myself, but moving the cursor to the right edge of the screen under Wayland caused the cursor to glitch out

**Comment by TheNathan0** [comment]
I take that nothing happens if you try moving it to the top left then.  Speaking of KDE, for some reason is Plasma 6 is more broken on T620 Mali GPUs than it was with Plasma 5.  In fact, at least on Wayland, the panels crash and you're immediately left with a strange buggy environment with a cursor and a black screen.  This was on the latest postmarketOS.  Gnome doesn't exhibit this issue however.  This is most probably some Panfrost bug since I haven't experienced this problem on any other Mali GPU running Panfrost.

Also, expect KDE performance to improve even more soon.  Plasma 6.1 will bring in triple buffering patches, which will make the desktop environment more performant in general, even on lower end GPUs.

**Comment by Salahd94** [comment]
Hi guys I sill can't install and run glmark2 on pmos can u write down the set of commands that works in pmos terminal 🙏

**Comment by jenneron** [comment]
@Salahd94 https://wiki.postmarketos.org/wiki/Terminal_cheat_sheet and https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper

**Comment by signalno11** [comment]
For me, the wifi refuses to work. I also tested it with my phone's hotspot, and it connects, but never gets the DNS or anything. Maybe DHCP doesn't work? It automatically disconnects from the network anyway. 

It fails with ip-config-unavailable and I also see a kernel error: "mwifiex_sdio_card_to_host: read iomem failed: -84"

This is a March '13 model, and I had to switch to the legacy kernel.

**Comment by signalno11** [comment]
> @signalno11 as mentioned [here](https://github.com/hexdump0815/imagebuilder/blob/main/systems/chromebook_snow/readme.md)
> 
> please be aware that there seems to be an early batch of snow chromebooks which are not running stable with a mainline kernel, but are running well with a legacy chromeos based kernel - looks like some hardware problem, which was silently fixed in the chromeos kernel but never made it to mainline - this problem has been observed on a snow chromebook from october 2012 and on one from january 2013, so i guess it affects the first ones from end of 2012 and beginning of 2013 - see: https://github.com/hexdump0815/imagebuilder/issues/63
> 
> at least this seams to be the issue

Yes, I've installed the old kernel, but the networking isn't working.

**Comment by signalno11** [comment]
The journal is filled with "mlan0: CTRL-EVENT-SCAN-FAILED ret=-16 retry=1" from wpa_supplicant. I also got bad firmware alerts before disabling power management.

**Comment by hexdump0815** [comment]
@signalno11 - maybe broken/dying hardware?

**Comment by signalno11** [comment]
> @signalno11 - maybe broken/dying hardware?

Yeah. I tried ChromeOS again and the wifi is locking up over there too. I presume there's no PCIe, so I'll probably get a USB wifi card for it.

**Comment by xboxboy** [comment]
I've installed and followed/used the post install scripts (nice work BTW). I've since installed UFW, but upon running "sudo ufw enable" I get various warnings mostly of 'missing kernel module?'
What's the best path here? Can I simply installed these modules, or am I needing to rebuild a kernel?

**Comment by thenameisluk** [comment]
both approaches are valid
if you look at the gentoo wiki you can see which modules are needed
https://wiki.gentoo.org/wiki/Ufw

**Comment by hexdump0815** [comment]
if you used the last official image from 2023, then maybe give the newer testimage a try as it has a newer kernel build with more kernel options enabled - see: https://github.com/velvet-os/imagebuilder-testimages/releases/tag/spring-release-2025-testimages-round-02

**Comment by xboxboy** [comment]
> both approaches are valid if you look at the gentoo wiki you can see which modules are needed https://wiki.gentoo.org/wiki/Ufw

Thanks, Interesting: Never dug that deep before.


> if you used the last official image from 2023, then maybe give the newer testimage a try as it has a newer kernel build with more kernel options enabled - see: https://github.com/velvet-os/imagebuilder-testimages/releases/tag/spring-release-2025-testimages-round-02

Thanks: Yes it was the latest image. I have used the test image: UFW installed and appears to be able to be configured and activated, thank you. I did have trouble opening synaptic and gufw from the GUI menu, but both start and run fine via CLI in the terminal emulator, I suspect that's more a base OS issue than anything from Velvet-OS?

Do I continue to use this test image, and it will update to stable when it's released, or will I have to wait for the new release?

Many thanks.

**Comment by thenameisluk** [comment]
if everything works on current image you won't need to change the images when stable one releases

apt will update stuff automatically when you run ```sudo apt update && sudo apt full-upgrade```
